test/checker/test_native_gate_checker.py

```diff
@@ -0,0 +1,55 @@
+# Tests for native gate checker pass
+
+import math
+
+import pytest
+
+from opensquirrel import CircuitBuilder
+from opensquirrel.circuit import Circuit
+from opensquirrel.ir import Float
+from opensquirrel.passes.checker.native_gate_checker import NativeGateChecker
+
+
+@pytest.fixture(name="checker")
+def checker_fixture() -> NativeGateChecker:
+    native_gate_set = ["H", "Z", "Y", "Rx", "CNOT"]
+    return NativeGateChecker(native_gate_set)
+
+
+@pytest.fixture
+def circuit_with_3_cnot() -> Circuit:
+    builder = CircuitBuilder(5)
+    builder.H(0)
+    builder.CNOT(0, 1)
+    builder.H(2)
+    builder.CNOT(1, 2)
+    builder.CNOT(2, 4)
+    builder.Z(2)
+    builder.Y(1)
+    builder.Rx(0, Float(math.pi / 3))
```

Comment:
`Float` type shouldn't be required.

```suggestion
    builder.Rx(0, math.pi / 3)
```